### ISSUE TITLE
Custom types broken after upgrading to ember-cli 2.3.0-beta.2

### ISSUE THREAD

**Issue by holandes22** [poteto/ember-cli-flash#127]
I have a configuration like so:

    flashMessageDefaults: {                                                                                                                                                      
      timeout: 4000,                                                                                                                                                             
      types: ['positive', 'negative']                                                                                                                                            
    }, 

After upgrading ember-cli, several acceptance tests fail since flashMessage service has undefined for the positive or negative methods. Is like the configuration is not being used.

I saw this in the ember-cli release notes
  
    #5321 Ensure EmberAddon.env() works the same as EmberApp.env() @jasonmit

which seems related. 

It is probably a regression in ember-cli as it was working fine at with 2.3.0-beta.1
but wanted to confirm before opening an issue over there.

**Comment by holandes22** [comment]
Is not as initially thought that my custom config is not being loaded, as I see that the timeout value I set is being used. The problem appears to be with the types value and the way they are merged with the deafaults.

I added some debug messages to console in the addon to see what config I get.

With the value for types ['positive', 'negative']. I get

```
...
 [ 'success',
     'info',
     'warning',
     'danger',
     'alert',
     'secondary',
     'e',
     'f' ],
...
```
